src/coll/mod.rs

```diff
@@ -297,6 +297,70 @@ impl Collection {
                                             CommandType::FindOneAndUpdate)
     }
 
+    /// Runs a "map-reduce" query on the collection.
+    ///
+    /// # Arguments
+    ///
+    /// `map` - A JavaScript function that selects the data to be transformed.
+    /// `reduce` - A JavaScript function that transforms the selected data
+    /// `out` - The name of the collection to store the output into.
+    pub fn map_reduce(&self, map: &str, reduce: &str, out: &str) -> Result<bson::Document> {
+        let cmd_doc = doc! {
+            "mapReduce" => (self.name()),
+            "map" => (Bson::JavaScriptCode(String::from(map))),
+            "reduce" => (Bson::JavaScriptCode(String::from(reduce))),
+            "out" => (String::from(out))
+        };
+
+        self.db.command(cmd_doc, CommandType::MapReduce, None)
+    }
+
+    /// Runs a "map-reduce" query on the collection.
+    ///
+    /// # Arguments
+    ///
+    /// `map` - A JavaScript function that selects the data to be transformed.
+    /// `reduce` - A JavaScript function that transforms the selected data
+    /// `options` - Other options for the map-reduce query.
+    pub fn map_reduce_with_options(&self, map: &str, reduce: &str, options: MapReduceOptions) -> Result<bson::Document> {
```

Comment:
when dealing with `{ "out" : { "inline" : 1} }`, some folks may prefer to have a `Result<Cursor>` if the document are still big enough to be fetched as `bson::Document`, just like aggregation did.